### ISSUE TITLE
Uncaught at saga thrown from render.

### ISSUE THREAD

**Issue by ashik94vc** [redux-saga/redux-saga#1004]
Okay, I found the most weird thing about saga today. I found that whenever there is an error in my view component which is rendered based on the result from redux, the error is not thrown in react but gets thrown through saga, like this. 

![error screen](http://i.imgur.com/Z5cCePx.png). 

The following error was caused by using an Android component in Ios. Though these errors should be thrown, they should be thrown by react and not by saga. I read the docs which said that the `put` doesn't bubble back errors to saga. If that's the case, why should saga throw the error when clearly the issue is with the render statement in my code.

**Comment by Andarist** [comment]
What version of `redux-saga` do u use? Could u show simplified code of `fetchTickets` and the component which causes this?

**Comment by ashik94vc** [comment]
the saga version I'm using is 0.15.3.
```
try {
        const payload = yield call(Api.fetchTickets);
        yield put(ticketFetchSucceeded(payload));
    } catch(error) {
        yield put(ticketFetchFailed());
    }

Assume that the component has a `<TouchableNativeFeedback/>` which ain't supported in iOS. The error should be thrown but not the messed up one react throws.

**Comment by Andarist** [comment]
is the error thrown into the saga? is it catchable? or does it only result in an error overlay in react native?

**Comment by drdaeman** [comment]
Got the same behavior today. I was renaming a variable in my component's `render` method and missed one. So, extremely oversimplified, my code was like this:

    @connect(
      state => ({}),
      dispatch => {fetchData: dispatch({type: "FETCH_FOO"})}
    )
    class Foo extends React.Component {
      constructor(props) {
        super(props)
        props.fetchData()
      }

      render() {
        return <Bar baz={noSuchVariable.uhhh} />  // Here is where it actually failed
      }
    }

However, the exception was long "`uncaught at rootSaga render@http.........`". I was a bit confused, as I've also touched my reducers at the same time, so it took me some time to notice the `render` and recognize it's where it really crashes.

There was a saga running:

    function* watchFetchFoo() {
      while (true) {
        try {
          yield take('FETCH_FOO')
          const data = yield call(() => fetch('/api/foo/').then(res => res.json()))
          yield put({type: 'FETCH_FOO_SUCCESS', payload: data})
        } catch (e) {
          console.log("Error in watchFetchFoo", e)  // No errors were reported here
        }
      }
    }

But no exceptions happened inside it (I've specifically added `try`...`catch` to make sure of this), and `rootSaga` was essentially just `yield all([watchFetchFoo()])` - I've commented out all the other sagas I've used.

It's not exactly a reproducible case - the code I wrote above is untested oversimplification of the one I have in a real project, but I hope it can be converted into one and help diagnosing this further.

**Comment by Andarist** [comment]
@drdaeman You are facing this issue https://github.com/redux-saga/redux-saga/issues/632

It's one of my top priorities to establish which behaviour is correct and to stay with it, however more I think about it I think we should fail soundly - so it will be a somewhat breaking change.

**Comment by ashik94vc** [comment]
I literally got close to pinpointing to the exact location where the problem arose. I'm quite busy now, but will try to narrow down to the place where this gets fuzzy and will try to attach it.

**Comment by Andarist** [comment]
Thanks for the help, additional info will be aprpeciated.

When I have time, I'm planning to wrap just **few minor** things up. Release last `0.x` version with new features and at the same time release `v1` - getting rid of deprecation warnings and *minor* breaking changes - this one will be one of them.

We'll come back to failing soundly in case of an error during `put`.

**Comment by Andarist** [comment]
I have started work on v1, gathering all things that need to be done before the release, you can track the progress [here](https://github.com/redux-saga/redux-saga/milestone/1).

In this release I will change the behaviour of `put` regarding the errors, it should not swallow uncaught errors.

Going to close this issue as it got stale and also its a duplicate of others.
